types/ember-data/test/relationships.ts

```diff
@@ -10,15 +12,19 @@ const Polymorphic = DS.Model.extend({
     paymentMethods: DS.hasMany('payment-method', { polymorphic: true })
 });
 
-const BlogPost = DS.Model.extend({
-    title: DS.attr('string'),
-    tag: DS.attr('string'),
-    comments: DS.hasMany('comment', { async: true }),
-    relatedPosts: DS.hasMany('post')
-});
+class Comment extends DS.Model {
+    author = DS.attr('string');
+}
 
-let blogPost = this.get('store').peekRecord('blog-post', 1);
+class BlogPost extends DS.Model {
+    title = DS.attr('string');
+    tag = DS.attr('string');
+    comments = DS.hasMany<Comment>('comment', { async: true });
+    relatedPosts = DS.hasMany('post');
+}
 
-blogPost.get('comments').then((comments: any[]) => {
+let blogPost = store.peekRecord<BlogPost>('blog-post', 1);
+blogPost!.get('comments').then((comments) => {
```

Comment:
*Huh.* I missed or forgot this bit in the TS 2.0 stuff; I've never actually seen it in the wild. (For others looking on: it's the [non-null assertion operator](https://github.com/Microsoft/TypeScript/wiki/What's-new-in-TypeScript#non-null-assertion-operator).